wgpu-hal/src/dx12/adapter.rs

```diff
@@ -284,9 +314,10 @@ impl super::Adapter {
                         _ => d3d12_ty::D3D12_MAX_SHADER_VISIBLE_SAMPLER_HEAP_SIZE,
                     },
                     // these both account towards `uav_count`, but we can't express the limit as as sum
-                    max_storage_buffers_per_shader_stage: base.max_storage_buffers_per_shader_stage,
-                    max_storage_textures_per_shader_stage: base
-                        .max_storage_textures_per_shader_stage,
+                    // of the two, so we divide it by 4 to account for the worst case scenario
+                    // (2 shader stages, with both using 16 storage textures and 16 storage buffers)
+                    max_storage_buffers_per_shader_stage: (uav_count / 4).min(8),
+                    max_storage_textures_per_shader_stage: (uav_count / 4).min(8),
```

Comment:
I think we shouldn't force the minimum to be 8 here.

> so I'm not sure if this should error if we're FL11.0 + tier 1

We will if the limits passed at device creation are not satisfied (`Limits::default()` correspond to WebGPU's minimum required limits).

https://github.com/gfx-rs/wgpu/blob/4478c52debcab1b88b80756b197dc10ece90dec9/wgpu-core/src/instance.rs#L347-L349

Reply:
So I should just remove the min and let it fail for devices that only support 8 UAVs?

Reply:
Yeah, as a user you can always lower the limits you pass to device creation if you really want the app to run on older hardware.